static/js/settings_org.js

```diff
@@ -813,18 +783,27 @@ function get_email_notification_batching_setting_element_value() {
     return setting_value_in_minutes * 60;
 }
 
-function get_message_edit_or_delete_limit_setting_value($input_elem) {
+function get_message_edit_or_delete_limit_setting_value($input_elem, for_api_data = true) {
     const select_elem_val = $input_elem.val();
 
     if (select_elem_val === "any_time") {
+        if (!for_api_data) {
+            return null;
+        }
         return JSON.stringify("unlimited");
     }
 
     if (select_elem_val !== "custom_period") {
         return Number.parseInt(select_elem_val, 10);
     }
 
-    return parse_time_limit($input_elem.parent().find(".admin-realm-time-limit-input"));
+    const custom_input_val = parse_time_limit(
+        $input_elem.parent().find(".admin-realm-time-limit-input"),
+    );
+    if (Number.isNaN(custom_input_val)) {
+        return null;
```

Comment:
A comment here would also be helpful.

Reply:
Also did one more change here which I will elaborate in the comment below.